cmake/sysbuild/partition_manager: nRF54L reserve UICR.OTP for users (#24686)
nrfconnect/sdk-nrf: doc/nrf/app_dev/device_guides/nrf54l/otp_map_nrf54l.rst

.. _ug_nrf54l_otp_map:

nRF54L OTP memory map for |NCS|
###############################

.. contents::
   :local:
   :depth: 2

The nRF54L Series devices are equipped with emulated OTP memory that can be used to store various assets which requires to be permanent and non-volatile.
This memory is part of the UICR memory space and can only be written to a non 0xFFFFFFFF value once after Erase All operation.
The OTP is organized into 4 bytes words.

.. list-table:: OTP memory map for nRF54L
    :header-rows: 1
    :align: center
    :widths: auto

    * - UICR.OTP[] Range
      - Size (4 bytes words)
      - Assignment
      - Description
    * - 0 - 287
      - 288
      - bl_storage space
      - Used for bootloaders and secure firmware storage realization.
    * - 288 - 319
      - 32
      - custom usage

>       - Available for users custom usage.

Reviewer: ```suggestion
      - Available for user's custom usage.
```

.. _ug_nrf54l_otp_map:

nRF54L OTP memory map for |NCS|
###############################

.. contents::
   :local:
   :depth: 2

The nRF54L Series devices are equipped with emulated One Time Programmable memory (OTP) that can be used to store various assets which requires to be permanent and non-volatile.

> This memory is part of the UICR memory space and can only be written to a non 0xFFFFFFFF value once after Erase All operation.

Reviewer 2: ```suggestion
This memory is a part of the User Information Configuration Register (UICR) memory space and can only be written to non-default values (non-0xFFFFFFFF) only once after an erase all operation.
```

The OTP is organized into 4 bytes words and are mapped as UICR.OTP[] array. Refer to the device Datasheet for more details.

> Usage of this memory is software defined and the |NCS| allocates a part of this memory for its own usage.

Reviewer 2: ```suggestion
The allocation of this memory is determined by the software, and the |NCS| reserves a part of this memory for its own usage.
```

.. list-table:: OTP memory assignment map for nRF54L Series devices
    :header-rows: 1
    :align: center
    :widths: auto

    * - UICR.OTP[] Range
      - Size (4 bytes words)
      - Assignment
      - Description
    * - 0 - 287
      - 288
      - bl_storage space
      - Used for bootloaders and secure firmware storage realization.
    * - 288 - 319
      - 32
      - custom usage

>       - Available for users custom usage.

Reviewer 2: ```suggestion
      - Custom usage
      - Available for custom user-defined purposes.
```

.. _ug_nrf54l_otp_map:

nRF54L One-Time Programmable memory map for |NCS|
#################################################

.. contents::
   :local:
   :depth: 2

The nRF54L Series devices are equipped with emulated One-Time Programmable (OTP) memory, designed to store assets permanently and in a non-volatile format.
This memory is a part of the User Information Configuration Register (UICR) memory space.
It can only be programmed with non-0xFFFFFFFF values, and it is possible only once after an erase all operation.
The OTP is organized into 4-byte words and is mapped as the UICR.OTP[] array.
For details, refer to the :ref:`datasheet<ug_nrf54l>` for the specific device model.

The allocation of this memory is determined by the software, and the |NCS| reserves a part of it for its own usage.

.. list-table:: OTP memory assignment map for nRF54L Series devices
    :header-rows: 1
    :align: center
    :widths: auto

    * - UICR.OTP[] Range
      - Size (4-byte words)
      - Assignment
      - Description
    * - 0 - 287
      - 288

>       - bl_storage space

Reviewer 2: ```suggestion
      - ``bl_storage`` space
```
